maxdiskusage.cc

```diff
@@ -79,7 +80,7 @@ maxdiskusage_notify(MYSQL_THD thd,
 
     if ((maxdiskusage_pct < 100) && (used_pct >= maxdiskusage_pct))
     {
-      if (strncmp(maxdiskusage_action, "WARN", 6) == 0)
+      if ((strncmp(maxdiskusage_action, "WARN", 6) == 0 || (strncmp(maxdiskusage_action, "WARN_AND_BLOCK", 6) == 0 && used_pct < maxdiskusage_block_pct))
```

Comment:
```c
#include <stdio.h>
#include <string.h>

int main() {
  if (strncmp("foobar", "foo", 6) == 0) {
    puts("foobar and foo match (6)");
  } else {
    puts("foobar and foo DO NOT match (6)");
  }

  if (strncmp("foobar", "foofoo", 6) == 0) {
    puts("foobar and foofoo match (6)");
  } else {
    puts("foobar and foofoo DO NOT match (6)");
  }

  if (strncmp("foobar", "foobar", 6) == 0) {
    puts("foobar and foobar match (6)");
  } else {
    puts("foobar and foobar DO NOT match(6)");
  }

  if (strncmp("foobar", "foofoo", 3) == 0) {
    puts("foobar and foofoo match (3)");
  } else {
    puts("foobar and foofoo DO NOT match (3)");
  }
}
```

```
$ cc -Wall foo.c -o foo
$ ./foo
foobar and foo DO NOT match (6)
foobar and foofoo DO NOT match (6)
foobar and foobar match (6)
foobar and foofoo match (3)
```
Be careful as `strncmp("WARN", "WARN_AND_BLOCK", 4)` returns 0. In this case 5 or more should be used to be able to differentiate.
```suggestion
      if ((strncmp(maxdiskusage_action, "WARN", 10) == 0 || (strncmp(maxdiskusage_action, "WARN_AND_BLOCK", 10) == 0 && used_pct < maxdiskusage_block_pct))
```